kotsadm/pkg/kurl/kurl_nodes.go

```diff
@@ -0,0 +1,148 @@
+}
+
+type NodeConditions struct {
+	MemoryPressure bool `json:"memoryPressure"`
+	DiskPressure   bool `json:"diskPressure"`
+	PidPressure    bool `json:"pidPressure"`
+	Ready          bool `json:"ready"`
+}
+
+// GetNodes will get a list of nodes with stats
+func GetNodes(client kubernetes.Interface) (*KurlNodes, error) {
+	nodes, err := client.CoreV1().Nodes().List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return nil, errors.Wrap(err, "list nodes")
+	}
+
+	mc, err := k8s.Metricsset()
+	if err != nil {
+		return nil, errors.Wrap(err, "get metrics client")
+	}
+
+	toReturn := KurlNodes{}
+
+	for _, node := range nodes.Items {
+		cpuCapacity := CapacityAvailable{}
+		memoryCapacity := CapacityAvailable{}
+		podCapacity := CapacityAvailable{}
+
+		memoryCapacity.Capacity = float64(node.Status.Capacity.Memory().Value()) / 1000000000 // capacity in GB
+
+		cpuCapacity.Capacity, err = strconv.ParseFloat(node.Status.Capacity.Cpu().String(), 64)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parse CPU capacity %q for node %s", node.Status.Capacity.Cpu().String(), node.Name)
+		}
+
+		podCapacity.Capacity, err = strconv.ParseFloat(node.Status.Capacity.Pods().String(), 64)
+		if err != nil {
+			return nil, errors.Wrapf(err, "parse pod capacity %q for node %s", node.Status.Capacity.Pods().String(), node.Name)
+		}
+
+		nodeMetrics, err := mc.MetricsV1beta1().NodeMetricses().Get(context.TODO(), node.Name, metav1.GetOptions{})
+		if err != nil {
+			logger.Infof("got error %s retrieving stats for node %s", err.Error(), node.Name)
+		} else {
+			memoryCapacity.Available = memoryCapacity.Capacity - (float64(nodeMetrics.Usage.Memory().Value()) / 1000000000)
+
+			cpuCapacity.Available, err = strconv.ParseFloat(nodeMetrics.Usage.Cpu().String(), 64)
+			if err != nil {
+				return nil, errors.Wrapf(err, "parse CPU available %q for node %s", nodeMetrics.Usage.Cpu().String(), node.Name)
+			}
+			cpuCapacity.Available = cpuCapacity.Capacity - cpuCapacity.Available
+
+			nodeMetrics.Usage.Pods()
+
+			podCapacity.Available, err = strconv.ParseFloat(nodeMetrics.Usage.Pods().String(), 64)
+			if err != nil {
+				return nil, errors.Wrapf(err, "parse pods available %q for node %s", nodeMetrics.Usage.Pods().String(), node.Name)
+			}
+			podCapacity.Available = podCapacity.Capacity - podCapacity.Available
+		}
+
+		toReturn.Nodes = append(toReturn.Nodes, Node{
+			Name:           node.Name,
+			IsConnected:    true,
+			CanDelete:      node.Spec.Unschedulable,
+			KubeletVersion: node.Status.NodeInfo.KubeletVersion,
+			CPU:            cpuCapacity,
+			Memory:         memoryCapacity,
+			Pods:           podCapacity,
+			Conditions:     findNodeConditions(node.Status.Conditions),
+		})
+	}
+
+	kurlConf, err := ReadConfigMap(client)
+	if err != nil {
+		if kuberneteserrors.IsNotFound(err) {
+			toReturn.IsKurlEnabled = false
+			return &toReturn, nil
+		}
+		return nil, errors.Wrap(err, "get kurl config")
+	}
+
+	toReturn.IsKurlEnabled = true
```

Comment:
I know this was ported from TS, but having `IsKurlEnabled` feels weird. Isn't this query only called if kurl is enabled? and the UI only shows the `Cluster Management` tab if kurl is enabled. https://github.com/replicatedhq/kots/blob/0117fba630a0c0b48959ea407564f549833f86f8/kotsadm/web/src/components/shared/NavBar.jsx#L168

Reply:
Unless we want to show some basic info for non-kurl clusters as well in the future.

Reply:
That was pretty much my thought
A lot of the data is available for kurl or non-kurl clusters